contrib/setup-z3.sh

```diff
@@ -1,13 +1,21 @@
 #!/bin/bash
 set -e
 
-Z3_VERSION=6cc52e04c3ea7e2534644a285d231bdaaafd8714
+Z3_VERSION=z3-4.12.6
 
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
 DEPS=$DIR/../deps
 
 mkdir -p $DEPS
 
+if [ "$(uname)" == "Darwin" ]; then
+    NUM_CORES=$(sysctl -n hw.logicalcpu)
+elif [ "$(expr substr $(uname -s) 1 5)" == "Linux" ]; then
+    NUM_CORES=$(nproc)
+else
+    NUM_CORES=1
+fi
+
```

Comment:
I am not sure about this, perhaps this can go to a separate PR, while giving the user the option to choose number of cores?

Reply:
Well, I am copying this from contrib/setup-cvc5.sh